middle_end/flambda/types/inlining/inlining_arguments.ml

```diff
@@ -14,65 +14,158 @@
+
+  let meet args1 args2 =
+    {
+      max_inlining_depth = min args1.max_inlining_depth args2.max_inlining_depth;
+      call_cost = Float.min args1.call_cost args2.call_cost;
+      alloc_cost = Float.min args1.alloc_cost args2.alloc_cost;
+      prim_cost = Float.min args1.prim_cost args2.prim_cost;
+      branch_cost = Float.min args1.branch_cost args2.branch_cost;
+      indirect_cost = Float.min args1.indirect_cost args2.indirect_cost;
+      small_function_size = min args1.small_function_size args2.small_function_size;
+      large_function_size = min args1.large_function_size args2.large_function_size;
+      threshold = Float.min args1.threshold args2.threshold
+    }
+
+
+  let cost_i (flag : Clflags.Int_arg_helper.parsed) ~round =
+    Clflags.Int_arg_helper.get ~key:round flag
+
+  let cost_f (flag : Clflags.Float_arg_helper.parsed) ~round =
+    Clflags.Float_arg_helper.get ~key:round flag
+
+  let create ~round = {
+    max_inlining_depth = !Clflags.Flambda.Expert.max_inlining_depth;
+    call_cost = cost_f !Clflags.inline_call_cost ~round;
+    alloc_cost = cost_f !Clflags.inline_alloc_cost ~round;
+    prim_cost = cost_f !Clflags.inline_prim_cost ~round;
+    branch_cost = cost_f !Clflags.inline_branch_cost ~round;
+    indirect_cost = cost_f !Clflags.inline_indirect_cost ~round;
+    small_function_size = cost_i !Clflags.inline_small_function_size ~round;
+    large_function_size = cost_i !Clflags.inline_large_function_size ~round;
+    threshold = cost_f !Clflags.inline_threshold ~round;
+  }
+end
+
+type t = Args.t Or_unknown.t
+
+let unknown = Or_unknown.Unknown
+
+let print ppf = Or_unknown.print Args.print ppf
 
 let max_inlining_depth = function
-  | Unknown -> 0
-  | O1 -> 1
-  | O2 -> 2
-  | O3 -> 3
-  | Custom t -> t.max_inlining_depth
-
-let equal t1 t2 =
-  max_inlining_depth t1 = max_inlining_depth t2
-
-let compress t =
-  match t with
-  | O1 | O2 | O3 | Unknown -> t
-  | t ->
-    if equal t O1 then O1
-    else if equal t O2 then O2
-    else if equal t O3 then O3
-    else t
-
-let meet_custom t1 t2 =
-  let max_inlining_depth =
-    if (max_inlining_depth t1) <= (max_inlining_depth t2) then
-      max_inlining_depth t1
-    else
-      max_inlining_depth t2
-  in
-  Custom { max_inlining_depth }
+  | Or_unknown.Unknown -> assert false
+  | Or_unknown.Known Args.{ max_inlining_depth; _ } ->
+    max_inlining_depth
 
-let meet t1 t2 =
+let call_cost = function
+  | Or_unknown.Unknown -> assert false
+  | Or_unknown.Known Args.{ call_cost; _ } ->
+    call_cost
+
+let alloc_cost = function
+  | Or_unknown.Unknown -> assert false
+  | Or_unknown.Known Args.{ alloc_cost; _ } ->
+    alloc_cost
+
+let prim_cost = function
+  | Or_unknown.Unknown -> assert false
+  | Or_unknown.Known Args.{ prim_cost; _ } ->
+    prim_cost
+
+let branch_cost = function
+  | Or_unknown.Unknown -> assert false
+  | Or_unknown.Known Args.{ branch_cost; _ } ->
+    branch_cost
+
+let indirect_cost = function
```

Comment:
This is presumably "indirect call cost"?  Let's rename it, even if it involves renaming the CLI flag.

Reply:
Renamed indirect_cost to indirect_call_cost accross the codebase (cli flag and in inlining_arguments.ml)